addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -929,18 +931,44 @@ export function createRelatedModels(modelDefs, modelClasses = {}, opts = {}) {
 
                 if (oldRecord && keepLocalRelation) {
                     for (const [field, value] of Object.entries(record)) {
-                        if (field === "id") {
+                        const params = getFields(model)[field];
+                        if (field === "id" || !params) {
                             continue;
                         }
 
-                        const params = getFields(model)[field];
-                        if (params && X2MANY_TYPES.has(params.type)) {
-                            value.push(
-                                ...oldRecord[field]
-                                    .filter((r) => typeof r.id === "string")
-                                    .map((r) => r.id)
+                        if (X2MANY_TYPES.has(params.type)) {
+                            const oldRel = oldRecord[field]
+                                .filter((r) => typeof r.id === "string")
+                                .map((r) => r.id);
+
+                            const existingInRaw = record[field].filter((id) =>
+                                models?.[params.relation]?.get(id)
                             );
-                            record[field] = ["set", value];
+
+                            if (existingInRaw.length > 0) {
+                                oldRel.push(...existingInRaw);
+                            }
+
+                            if (oldRel.length > 0) {
+                                record[field] = [["set", oldRel]];
```

Comment:
```suggestion
                                record[field] = [["set", ...oldRec.map(id => models[model].get('id')).filter(Boolean)];
```

Reply:
I think old records will always be there, did u test it ?